src/renderer/components/pool/PoolChart.tsx

```diff
@@ -1,17 +1,169 @@
-import React from 'react'
+import React, { useMemo, useState } from 'react'
 
-import * as A from 'antd'
+import { baseAmount, baseToAsset, bnOrZero } from '@xchainjs/xchain-util'
+import { currencySymbolByAsset } from '@xchainjs/xchain-util'
+import BigNumber from 'bignumber.js'
+import * as FP from 'fp-ts/lib/function'
+import * as O from 'fp-ts/lib/Option'
 
+import { SelectedPricePoolAsset } from '../../services/midgard/types'
+import Chart from '../uielements/chart'
+import { ChartData, ChartDetail, ChartValues } from '../uielements/chart/types'
 import * as Styled from './PoolChart.style'
+import { LiquidityData, VolumeData } from './types'
 
 export type Props = {
+  selectedPricePoolAsset: SelectedPricePoolAsset
   isLoading?: boolean
+  priceRatio: BigNumber
+  volumeAllTimeData: O.Option<VolumeData[]>
+  volumeWeekData: O.Option<VolumeData[]>
+  liquidityAllTimeData: O.Option<LiquidityData[]>
+  liquidityWeekData: O.Option<LiquidityData[]>
```

Comment:
actually as i can see this values are the same at all...
types
```ts
export type VolumeData = {
  time: number
  poolVolume: string
}

export type LiquidityData = {
  time: number
  runeDepth: string
}
```
can be converted to utilized interface like
```ts
type ChartItem = {
  time: number; // which is actually timeStamp...
  value: string;
}
```

the second thing is why do we need all of this separation of props ?
the main question here is what will we do in case we want to display more data at the chart ? pass over and over new pairs (for week and all time periods) again and again ? and what if there are more time periods to select ? day/week/month/3 months/ 6 months ? really dont like this...
from my point of view `PoolChart`(and `DefaultChart`) should receive only values the need at the moment like:
```ts
type ChartProps = {
  isLoading: boolean
  displayType: 'line' | 'bar'
  charData: ChartItem[]
  timePeriods: string[] // or ('all' | 'week')[]
  setTimePeriod: (period: string) => void
  dataTypes: string[] // Liquidity and Volume
  setDataType: (dataType: string) => void
}
```
with ^^^this^^^ we can reuse such chart component anywhere we wanted to and decide which exact data we want to display on demand

so with such changes (if they are approved) we just need to place calculation and storing of all this data to the View-layer.
actually it also might be utilized. for `PoolChartView.tsx` might be something like this (this is not a final code. just main idea):

```ts
const availableTimePeriods = ['week', 'all'] as const
type TimePeriod = typeof availableTimePeriods[number]
const availableDataTypes = ['liquidity',  'volume'] as const
type DataType = typeof availableDataTypes[number]
type DataRequestParams = { period: TimePeriod; dataType: DataType }
  const savedParams = useRef<DataRequestParams>({
    period: availableTimePeriods[0],
    dataType: availableDataTypes[0]
  })
  const [chartDataRD, updateChartData] = useObservableState<RD.RemoteData<any, any>, Partial<A>>(
    (params$) =>
      FP.pipe(
        params$,
        RxOp.map((params) => (savedParams.current = { ...savedParams.current, ...params })),
        RxOp.switchMap((params) => {
          const getDataStream$ = params.dataType === 'liquidity' ? getDepthHistory$ : getSwapHistory$
// ^^^  getDepthHistory$ and getSwapHistory$  getters should be modified to return a single data interfaces ^^^
          const requestParams = params.period === 'all' ? {} : { from: weekAgoTime, to: curTime }

          return getDataStream$({
            ...requestParams,
            interval: 'day' as const
          })
        }),
        RxOp.map(/* map data as you need to */)
      ),
    RD.initial
  )

  const setTimePeriod = useCallback(
    (period: TimePeriod) => {
      updateChartData({ period })
    },
    [updateChartData]
  )

  const setDataTypePeriod = useCallback(
    (dataType: DataType) => {
      updateChartData({ dataType, period: availableTimePeriods[0] })
    },
    [updateChartData]
  )
```

so in this case for adding any new `DataType` or `TimePeriod` we just need to configure appropriate constants and modify appropriate getters... and Chart component will take ONLY data it needs to display....

@veado  what do you think ?

Reply:
Fixed by #1261